make/autoconf/lib-hsdis.m4

```diff
@@ -251,10 +253,31 @@ AC_DEFUN([LIB_SETUP_HSDIS_BINUTILS],
         HSDIS_CFLAGS="$HSDIS_CFLAGS -I$BINUTILS_DIR/include -I$BINUTILS_DIR/bfd"
       fi
       HSDIS_LDFLAGS=""
-      HSDIS_LIBS="$BINUTILS_DIR/bfd/libbfd.a $BINUTILS_DIR/opcodes/libopcodes.a $BINUTILS_DIR/libiberty/libiberty.a $BINUTILS_DIR/zlib/libz.a"
+      HSDIS_LIBS="$BINUTILS_DIR/bfd/.libs/libbfd.a $BINUTILS_DIR/opcodes/libopcodes.a $BINUTILS_DIR/libiberty/libiberty.a $BINUTILS_DIR/zlib/libz.a"
+      # If we have libsframe add it.
+      if test -e $BINUTILS_DIR/libsframe/.libs/libsframe.a; then
+        HSDIS_LIBS="$BINUTILS_DIR/bfd/.libs/libbfd.a $BINUTILS_DIR/opcodes/libopcodes.a $BINUTILS_DIR/libiberty/libiberty.a $BINUTILS_DIR/zlib/libz.a $BINUTILS_DIR/libsframe/.libs/libsframe.a"
+      fi
     fi
   fi
 
+  AC_MSG_CHECKING([Checking binutils API])
+  if test -n "$BINUTILS_SRC"; then
+    include_dis="#include \"$BINUTILS_SRC/include/dis-asm.h\""
```

Comment:
All this quoting and duplication seems quite messy. Can't you just set e.g. `binutils_include_dir` to either `%BINUTILS_SRC/include` or `%BINUTILS_DIR/include`, and then just do 

```
AC_LANG_PROGRAM([#include "$binutils_include_dir/dis-asm.h"], ...
```
?

(Code written in github editor; warning for quoting mistakes etc)

Reply:
I can check. No really my usual place in the world doing autoconf :)